SYCL/ESIMD/api/hardware_dispatch.cpp

```diff
@@ -0,0 +1,81 @@
+//==---------------- hardware_dispatch.cpp  - ESIMD hardware dispatch test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// UNSUPPORTED: esimd_emulator
+// RUN: %clangxx -fsycl -fsycl-targets=intel_gpu_bdw %s -o %t.out
+// RUN: %t.out
```

Comment:
using intel_gpu_bdw here means that the test will pass only on BDW, and fail on all non-BDW CI machines.
If that is correct understanding, then the test needs some re-work to work properly always.

Reply:
It may not fail with SegFault-like error, but what is the expected value written to result array. Please see this comment: https://github.com/intel/llvm-test-suite/pull/1486/files#r1061925275